[NB] Implicit systems of size one generate assingments instead of residuals

If you run this model with `--newBackend`

```modelica
model MWE
  parameter Integer n = 10;
  Real x[n](each start = 1.0);
equation
  for i in 1:n loop
    x[i] * x[i] = x[i];
  end for;
end MWE;
```
it generates the wrong code with error message

```
MWE_06inz.c: In function ‘MWE_eqFunction_1’:
MWE_06inz.c:32:229: error: lvalue required as left operand of assignment
   32 |       ((&data->localData[0]->realVars[data->simulationInfo->realVarsIndex[0]] /* x[1] variable */)[_omcQ_24i1 - 1]) * ((&data->localData[0]->realVars[data->simulationInfo->realVarsIndex[0]] /* x[1] variable */)[_omcQ_24i1 - 1]) = (&data->localData[0]->realVars[data->simulationInfo->realVarsIndex[0]] /* x[1] variable */)[_omcQ_24i1 - 1];
      |                                                                                                                                                                                                                                     ^
```

Basically algebraic loops of size one are the problem. If I add another equation it works correctly. The evil part is, if the lhs is a cref you don't notice the error!

--- a/OMCompiler/Compiler/NBackEnd/Modules/3_Post/NBTearing.mo
+++ b/OMCompiler/Compiler/NBackEnd/Modules/3_Post/NBTearing.mo
@@ -186,6 +186,16 @@ public
           status  = NBSolve.Status.IMPLICIT);
       then finalize(new_comp, dummy, funcTree, index, VariablePointers.empty(), EquationPointers.empty(), Pointer.create(0), kind);
 
+      case StrongComponent.RESIZABLE_COMPONENT() algorithm
+        new_comp := StrongComponent.ALGEBRAIC_LOOP(
+          idx     = index,
+          strict  = singleImplicit(Slice.getT(comp.var), Slice.getT(comp.eqn)),
+          casual  = NONE(),
+          linear  = false,
+          mixed   = false,
+          status  = NBSolve.Status.IMPLICIT);
+      then finalize(new_comp, dummy, funcTree, index, VariablePointers.empty(), EquationPointers.empty(), Pointer.create(0), kind);
+
       // do nothing otherwise
       else (comp, dummy, funcTree, index);
     end match;

--- a/testsuite/simulation/modelica/NBackend/tearing/Makefile
+++ b/testsuite/simulation/modelica/NBackend/tearing/Makefile
@@ -2,6 +2,7 @@ TEST = ../../../../rtest -v
 
 TESTFILES = \
 minimalTearing_simple.mos \
+loopSizeOne.mos \
 
 # test that currently fail. Move up when fixed.
 # Run make failingtest

--- a/testsuite/simulation/modelica/NBackend/tearing/loopSizeOne.mos
+++ b/testsuite/simulation/modelica/NBackend/tearing/loopSizeOne.mos
@@ -1,0 +1,40 @@
+// name: LoopSizeOne
+// keywords: tearing, edge case
+// status: correct
+// teardown_command: rm -rf LoopSizeOne* _LoopSizeOne* output.log
+// cflags: --newBackend
+//
+// Tearing with loop of size 1.
+
+loadString("
+model LoopSizeOne
+  parameter Integer n = 3;
+  Real x[n](each start = 1.0);
+equation
+  for i in 1:n loop
+    x[i] = x[i] * x[i] - i;
+  end for;
+end LoopSizeOne;
+"); getErrorString();
+
+simulate(LoopSizeOne, stopTime=0.0); getErrorString();
+
+val(x[1], 0.0);
+val(x[2], 0.0);
+val(x[3], 0.0);
+
+// Result:
+// true
+// ""
+// record SimulationResult
+//     resultFile = "LoopSizeOne_res.mat",
+//     simulationOptions = "startTime = 0.0, stopTime = 0.0, numberOfIntervals = 500, tolerance = 1e-6, method = 'dassl', fileNamePrefix = 'LoopSizeOne', options = '', outputFormat = 'mat', variableFilter = '.*', cflags = '', simflags = ''",
+//     messages = "LOG_SUCCESS       | info    | The initialization finished successfully without homotopy method.
+// LOG_SUCCESS       | info    | The simulation finished successfully.
+// "
+// end SimulationResult;
+// ""
+// 1.6180339887498948
+// 2.0
+// 2.3027756377319946
+// endResult

